### PR TITLE
Fixes to make ekg-core compile against GHC 7.10.

### PR DESCRIPTION
**Pull request by seanparsons**
All I've really done is bump a few upper bounds on dependencies so that it compiles.

Probably the biggest change is that I used the mutli-ghc-travis project as a template for a Travis config, which helped me to test the varying versions of GHC, feel free to fire that into the sun if you so wish.

### CODE REVIEW

**Comment by seanparsons** [comment]
I have no idea how this has merge conflicts however...

File: ekg-core.cabal [hunk 1]
```diff
@@ -28,10 +28,10 @@ library
     System.Metrics.ThreadId
 
   build-depends:
-    ghc-prim < 0.4,
-    base >= 4.5 && < 4.8,
+    ghc-prim < 0.5,
+    base >= 4.5 && < 4.10,
```
Comment (tibbe) [inline ekg-core.cabal]: This should be 4.9.

**Comment by tibbe** [comment]
Would you mind fixing the one comment, squash the commits and force-push to the branch? Thanks!

File: ekg-core.cabal (hunk 1 above)
Comment (seanparsons) [inline ekg-core.cabal]: Isn't that bound what allows it to be built against HEAD? I guess you don't want something to be "compatible" but not eventually with the final version.

**Comment by seanparsons** [comment]
Ah, I see the issue, I still had an old fork and hadn't pulled the upstream changes which included a lot of the stuff I'd done in my branch. Oh well!

**Pull request merged**